eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -269,31 +269,27 @@ object Graph {
+  private def addEdgeWeight(sender: PublicKey, edge: GraphEdge, prev: RichWeight, currentBlockHeight: Long, weightRatios: WeightRatios, includeLocalChannelCost: Boolean): RichWeight = {
     val totalCost = if (edge.desc.a == sender && !includeLocalChannelCost) prev.cost else addEdgeFees(edge, prev.cost)
     val fee = totalCost - prev.cost
+    val hopCost = nodeFee(weightRatios.hopCostBase, weightRatios.hopCostMillionths, prev.cost)
     val totalCltv = if (edge.desc.a == sender && !includeLocalChannelCost) prev.cltv else prev.cltv + edge.update.cltvExpiryDelta
-    val factor = weightRatios match {
-      case None =>
-        1.0
-      case Some(wr) =>
-        import RoutingHeuristics._
+    import RoutingHeuristics._
 
-        // Every edge is weighted by funding block height where older blocks add less weight. The window considered is 1 year.
-        val channelBlockHeight = ShortChannelId.coordinates(edge.desc.shortChannelId).blockHeight
-        val ageFactor = normalize(channelBlockHeight, min = (currentBlockHeight - BLOCK_TIME_ONE_YEAR).toDouble, max = currentBlockHeight.toDouble)
+    // Every edge is weighted by funding block height where older blocks add less weight. The window considered is 1 year.
+    val channelBlockHeight = ShortChannelId.coordinates(edge.desc.shortChannelId).blockHeight
+    val ageFactor = normalize(channelBlockHeight, min = (currentBlockHeight - BLOCK_TIME_ONE_YEAR).toDouble, max = currentBlockHeight.toDouble)
 
-        // Every edge is weighted by channel capacity, larger channels add less weight
-        val edgeMaxCapacity = edge.capacity.toMilliSatoshi
-        val capFactor = 1 - normalize(edgeMaxCapacity.toLong.toDouble, CAPACITY_CHANNEL_LOW.toLong.toDouble, CAPACITY_CHANNEL_HIGH.toLong.toDouble)
+    // Every edge is weighted by channel capacity, larger channels add less weight
+    val edgeMaxCapacity = edge.capacity.toMilliSatoshi
+    val capFactor = 1 - normalize(edgeMaxCapacity.toLong.toDouble, CAPACITY_CHANNEL_LOW.toLong.toDouble, CAPACITY_CHANNEL_HIGH.toLong.toDouble)
 
-        // Every edge is weighted by its cltv-delta value, normalized
-        val cltvFactor = normalize(edge.update.cltvExpiryDelta.toInt, CLTV_LOW, CLTV_HIGH)
+    // Every edge is weighted by its cltv-delta value, normalized
+    val cltvFactor = normalize(edge.update.cltvExpiryDelta.toInt, CLTV_LOW, CLTV_HIGH)
 
-        // NB we're guaranteed to have weightRatios and factors > 0
-        (cltvFactor * wr.cltvDeltaFactor) + (ageFactor * wr.ageFactor) + (capFactor * wr.capacityFactor)
-    }
-    val totalWeight = prev.weight + fee.toLong * factor
+    // NB we're guaranteed to have weightRatios and factors > 0
+    val factor = weightRatios.biasFactor + (cltvFactor * weightRatios.cltvDeltaFactor) + (ageFactor * weightRatios.ageFactor) + (capFactor * weightRatios.capacityFactor)
+    val totalWeight = prev.weight + (fee + hopCost).toLong * factor
```

Comment:
Intuitively, I would have chosen a slightly different formula, where I would have kept that new hop factor separate from the other factors:
```
val totalWeight = prev.weight + hopCost * weightRatios.biasFactor + fee * ((cltvFactor * weightRatios.cltvDeltaFactor) + (ageFactor * weightRatios.ageFactor) + (capFactor * weightRatios.capacityFactor))
```

What do you think are the pros/cons of each formula? Is my suggestion dumb?

Reply:
You've misunderstood what the `biasFactor` does, it is not related to the `hopCost`. It biases towards just using the fee as the weight.

Reply:
Got it, the formula makes much more sense now! That explains why mine was different, since I was trying to apply a ratio to the `hopCost`.